src/main/java/org/altbeacon/beacon/Identifier.java

```diff
@@ -14,16 +16,15 @@
  */
 public class Identifier implements Comparable<Identifier> {
     private static final Pattern HEX_PATTERN = Pattern.compile("^0x[0-9A-Fa-f]*$");
-    private static final Pattern DECIMAL_PATTERN = Pattern.compile("^[0-9]+$");
-    private static final Pattern UUID_PATTERN = Pattern.compile("^[0-9A-Fa-f]{8}-?[0-9A-Fa-f]{4}-?[0-9A-Fa-f]{4}-?[0-9A-Fa-f]{4}-?[0-9A-Fa-f]{12}$");
-    private static final char[] hexArray = "0123456789abcdef".toCharArray();
-    private static final BigInteger maxInteger = BigInteger.valueOf(65535);
```

Comment:
`65535 < Integer.MAX_VALUE` can be assumed nearly everywhere. Therefore `BigInteger` is not needed and we can use an ordinary `int`.